examples/api-tests/src/views.spec.js

```diff
@@ -53,8 +53,8 @@ describe('Views', function () {
 
             view = await contribution.toggleView();
             assert.notEqual(view, undefined);
-            assert.isUndefined(shell.getAreaFor(view));
-            assert.isUndefined(shell.getTabBarFor(view));
+            assert.equal(shell.getAreaFor(view), contribution.defaultViewOptions.area);
+            assert.isDefined(shell.getTabBarFor(view));
             assert.isFalse(view.isVisible);
             assert.notEqual(view, shell.activeWidget);
```

Comment:
toggle activates a widget by default

@spoenemann btw did you figure out how to run and debug tests locally? if not https://github.com/eclipse-theia/theia/blob/master/doc/api-testing.md#running-tests

Reply:
Yes, but I don't understand the test failure.

```
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module '../lib/theia'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/workspace/theia/dev-packages/cli/bin/theia:2:1)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
```

Reply:
On Travis the output is different:

```
@theia/example-browser: Error: Error: unexpected property: Symbol(Symbol.toStringTag)\n    at Object.get (http://127.0.0.1:3000/bundle.js:121035:19)\n    at Proxy.toString (<anonymous>)\n    at type (/home/travis/build/eclipse-theia/theia/node_modules/mocha/mocha.js:7593:6)\n    at Object.canonicalize (/home/travis/build/eclipse-theia/theia/node_modules/mocha/mocha.js:7761:26)\n    at /home/travis/build/eclipse-theia/theia/node_modules/mocha/mocha.js:7805:45\n    at Array.forEach (<anonymous>)\n    at /home/travis/build/eclipse-theia/theia/node_modules/mocha/mocha.js:7804:12\n    at withStack (/home/travis/build/eclipse-theia/theia/node_modules/mocha/mocha.js:7764:5)\n    at Object.canonicalize (/home/travis/build/eclipse-theia/theia/node_modules/mocha/mocha.js:7801:7)\n    at /home/travis/build/eclipse-theia/theia/node_modules/mocha/mocha.js:7805:45
4552@theia/example-browser: Error: Error: Request 'inspect' failed\n    at Proxy.<anonymous> (http://127.0.0.1:3000/bundle.js:141304:33)\n    at formatValue (/home/travis/build/eclipse-theia/theia/node_modules/chai/chai.js:8462:21)\n    at formatProperty (/home/travis/build/eclipse-theia/theia/node_modules/chai/chai.js:8704:15)\n    at /home/travis/build/eclipse-theia/theia/node_modules/chai/chai.js:8596:14\n    at Array.map (<anonymous>)\n    at formatValue (/home/travis/build/eclipse-theia/theia/node_modules/chai/chai.js:8595:19)\n    at formatProperty (/home/travis/build/eclipse-theia/theia/node_modules/chai/chai.js:8704:15)\n    at /home/travis/build/eclipse-theia/theia/node_modules/chai/chai.js:8596:14\n    at Array.map (<anonymous>)\n    at formatValue (/home/travis/build/eclipse-theia/theia/node_modules/chai/chai.js:8595:19)
4553@theia/example-browser: Error: Error: Request 'inspect' failed\n    at Proxy.<anonymous> (http://127.0.0.1:3000/bundle.js:141304:33)\n    at formatValue (/home/travis/build/eclipse-theia/theia/node_modules/chai/chai.js:8462:21)\n    at formatProperty (/home/travis/build/eclipse-theia/theia/node_modules/chai/chai.js:8704:15)\n    at /home/travis/build/eclipse-theia/theia/node_modules/chai/chai.js:8596:14\n    at Array.map (<anonymous>)\n    at formatValue (/home/travis/build/eclipse-theia/theia/node_modules/chai/chai.js:8595:19)\n    at formatProperty (/home/travis/build/eclipse-theia/theia/node_modules/chai/chai.js:8704:15)\n    at /home/travis/build/eclipse-theia/theia/node_modules/chai/chai.js:8596:14\n    at Array.map (<anonymous>)\n    at formatValue (/home/travis/build/eclipse-theia/theia/node_modules/chai/chai.js:8595:19)
```

Reply:
It seems that notequal does deep comparison and expect some object to implement such property. Could you try to replace it shallow compare?

Also I’m wrong test is correct a widget should be closed and it cannot be active. Toggle resolves too early.